src/csharp/Microsoft.Spark/Utils/PythonSerDe.cs

```diff
@@ -49,6 +53,7 @@ internal static object[] GetUnpickledObjects(Stream stream, int messageLength)
                 object unpickledItems = unpickler.loads(
                     new ReadOnlyMemory<byte>(buffer, 0, messageLength), 
                     stackCapacity: 102); // Spark sends batches of 100 rows, and +2 is for markers.
+                s_rowConstructor.Reset();
```

Comment:
this needs to be thread-safe no? note that this can be shared in daemon server.

Reply:
This is now thread-safe.